osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs

```diff
@@ -112,21 +114,36 @@ private void onJudgementLoaded(DrawableOsuJudgement judgement)
         }
 
         [BackgroundDependencyLoader(true)]
-        private void load(OsuRulesetConfigManager config)
+        private void load(OsuRulesetConfigManager config, IBeatmap beatmap)
         {
             config?.BindWith(OsuRulesetSetting.PlayfieldBorderStyle, playfieldBorder.PlayfieldBorderStyle);
 
-            RegisterPool<HitCircle, DrawableHitCircle>(10, 100);
+            var osuBeatmap = (OsuBeatmap)beatmap;
 
-            RegisterPool<Slider, DrawableSlider>(10, 100);
-            RegisterPool<SliderHeadCircle, DrawableSliderHead>(10, 100);
-            RegisterPool<SliderTailCircle, DrawableSliderTail>(10, 100);
-            RegisterPool<SliderTick, DrawableSliderTick>(10, 100);
-            RegisterPool<SliderRepeat, DrawableSliderRepeat>(5, 50);
+            RegisterPool<HitCircle, DrawableHitCircle>(20, 100);
+
+            // handle edge cases where a beatmap has a slider with many repeats.
+            int maxRepeatsOnOneSlider = 0;
+            int maxTicksOnOneSlider = 0;
+
+            if (osuBeatmap != null)
```

Comment:
I'm not sure this can actually be null in practice, but it's much of a muchness in this case.

Reply:
It happened tests which is why I added it :(